### ISSUE TITLE
Submit hangs indefinitely if there are validation errors

### ISSUE THREAD

**Issue by juvester** [testmycode/tmc-core#109]
SubmissionResultParser fails to parse the submissionStatus JSON if there are validation errors. This seems to cause an infinite loop in Submit.call() method where it just fetches the same JSON string over and over again. No interrupt exception is ever thrown, we let it run for 30+ mins.

Here's the [parse exception and the JSON string](https://gist.github.com/juvester/8e408835e9b2692c8251e3e03ffdc014) causing it. The same exception is thrown like every two seconds. Very easy to reproduce, just have incorrect indentation in some exercise that does checkstyle validations and do TmcCore.submit(). Local validation works fine.

@Salmela got the parsing to work by defining [ValidationResultImpl.validationErrors](https://github.com/testmycode/tmc-core/blob/e727226f19a55f596d9f635598f9612796e08d40/src/main/java/fi/helsinki/cs/tmc/core/domain/submission/ValidationResultImpl.java#L14) map as 'transient' but wasn't quite sure whether that's the right way to fix it or if it causes problems elsewhere.

Hope this helps!

**Comment by cxcorp** [comment]
Can confirm.

Snippet from response JSON:
```json
"validations": {
  "strategy": "FAIL",
  "validationErrors": {
    "Main.java": [
    {
      "column": 0,
      "line": 13,
      "message": "'if' construct must use curly braces '{}'.",
      "sourceName": "com.puppycrawl.tools.checkstyle.checks.blocks.NeedBracesCheck"
    }
    ]
  }
}
```

Here the parsing fails at the start of `"Main.java"`.

The problem is that class [`ValidationResultImpl`](https://github.com/testmycode/tmc-core/blob/784fe8d8e05a62e8e1f74d075181a5e7b474c7a9/src/main/java/fi/helsinki/cs/tmc/core/domain/submission/ValidationResultImpl.java#L14) keeps validation errors in a `Map<File, List<ValidationError>>`, however, `File` does not get serialized into just a string containing the filename. It gets serialized into (and deserialized from) an object like such:
```json
{"path":"Main.java"}
```

**Comment by nygrenh** [comment]
Fixed by #116
